client/components/auto-direction/index.jsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import userModule from 'lib/user';
+import { stripHTML } from 'lib/formatting';
+
+const user = userModule();
+
+/***
+ * Gets text content from react element in case that's a leaf element
+ * @param {React.Element} reactElement react element
+ * @returns {string|null} returns a text content of the react element or null if it's not a leaf element
+ */
+const getContent = ( reactElement ) => {
+	if ( ! reactElement ) {
+		return null;
+	}
+	const { props } = reactElement;
+
+	if ( ! props ) {
+		return null;
+	}
+
+	// The child is a text node
+	if ( typeof props.children === 'string' ) {
+		return props.children;
+	}
+
+	// This child has it's content set to external HTML
+	if ( typeof props.dangerouslySetInnerHTML === 'object' ) {
+		// Strip tags because we're only interested in the text, not markup
+		return props.dangerouslySetInnerHTML.__html
+			? stripHTML( props.dangerouslySetInnerHTML.__html )
```

Comment:
i'm ...minorly worried .. about this. we have some pretty big contents in the reader and copying the entire thing isn't going to be cheap. maybe we truncate it to our max examine * 2 before stripping?